### PR TITLE
Deploy Garble Guide using Mdbook to GH Pages

### PR DESCRIPTION
**Pull request by fkettelhoit**
Deployed website/book is here: https://sine-fdn.github.io/garble-lang/

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,11 +1,9 @@
...
-
-All programs written in Garble are deliberately Turing-incomplete (only supporting bounded recursion), guaranteeing that they can be compiled to circuits using only `AND`, `XOR` and `NOT` gates (without any kind of stateful latches or registers). Here's an example of solving the [Millionaire's Problem](https://en.wikipedia.org/wiki/Yao%27s_Millionaires%27_problem) in Garble:
+Garble is a simple programming language for [**Multi-Party Computation**](https://en.wikipedia.org/wiki/Secure_multi-party_computation) with [**Garbled Circuits**](https://en.wikipedia.org/wiki/Garbled_circuit). Garble programs are **compiled to boolean circuits** and always terminate. Garble is **statically typed, low-level, purely functional** and uses a **Rust-like syntax**. Garble is much simpler than Rust though, making it easy to learn and simple to [integrate](https://sine-fdn.github.io/garble-lang/integration.html) into MPC engines.
```
Comment (kisakishy) [inline README.md:3]: Garble is a simple programming language that enables describing functionalities for [**Secure Multi-Party Computation (SMPC or MPC in short)**](https://en.wikipedia.org/wiki/Secure_multi-party_computation) with **Boolean Circuits**.  The most prominent solution is based on [Garbled Circuits](https://en.wikipedia.org/wiki/Garbled_circuit).

I know I am stressing this but Garble could be used for any GMW protocol-based (secret sharing but with bools) MPC engine as well! It was a very prominent protocol at some point and Garble is definitely nowhere tied to Garbled circuits, but is tied to Boolean circuits.

Comment (kisakishy) [reply README.md:3]: Boolean circuit is generally always written capitalized, so "Garble programs are **compiled to Boolean circuits** and always terminate."

bool/boolean as a type is also written without capitalization for consistency, but Boolean circuits, Boolean grammar, Boolean algebra are not, due to coming from George Boole.

File: garble_docs/src/guide/intro.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# A Guide to Garble
+
+Garble is a programming language for [MPC](<(https://en.wikipedia.org/wiki/Secure_multi-party_computation)>) that is quite easy to learn and mostly works like Rust, except that it is much simpler and only implements a subset of Rust (without the borrow checker). The following guide introduces the main features of Garble.
+
+A minimal Garble program is a public function (conventionally called `main`), with each input conceptually belonging to a different party in a [MPC Garbled Circuit](https://en.wikipedia.org/wiki/Garbled_circuit). For example, the following program computes the boolean `and` of two parties:
```
Comment (kisakishy) [inline garble_docs/src/guide/intro.md:5]: ... in a Boolean circuit-based MPC protocol, e.g., the one using a [Garbled Circuit](https://en.wikipedia.org/wiki/Garbled_circuit). For example, the following program computes the boolean AND ($\wedge$) of two parties:

File: garble_docs/src/contributing.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+2. [`parse.rs`](https://github.com/sine-fdn/garble-lang/blob/main/src/parse.rs) parses a [`token::Token`](https://github.com/sine-fdn/garble-lang/blob/main/src/token.rs) sequence into an untyped [`ast::Program`](https://github.com/sine-fdn/garble-lang/blob/main/src/ast.rs).
+3. [`check.rs`](https://github.com/sine-fdn/garble-lang/blob/main/src/check.rs) type-checks an untyped [`ast::Program`](https://github.com/sine-fdn/garble-lang/blob/main/src/ast.rs), returning a typed [`ast::Program`](https://github.com/sine-fdn/garble-lang/blob/main/src/ast.rs).
+4. [`compile.rs`](https://github.com/sine-fdn/garble-lang/blob/main/src/compile.rs) converts a well-typed [`ast::Program`](https://github.com/sine-fdn/garble-lang/blob/main/src/ast.rs) into a [`circuit::Circuit`](https://github.com/sine-fdn/garble-lang/blob/main/src/circuit.rs).
+5. [`eval.rs`](https://github.com/sine-fdn/garble-lang/blob/main/src/eval.rs) executes a [`circuit::Circuit`](https://github.com/sine-fdn/garble-lang/blob/main/src/circuit.rs) with locally supplied inputs.
```
Comment (kisakishy) [inline garble_docs/src/contributing.md:13]: Maybe we could add: "This execution is for testing the correctness of the result locally and does not use any privacy-preserving techniques."

File: garble_docs/src/landing.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# The Garble Programming Language
+
+Garble is a simple programming language for [**Multi-Party Computation**](https://en.wikipedia.org/wiki/Secure_multi-party_computation) with [**Garbled Circuits**](https://en.wikipedia.org/wiki/Garbled_circuit). Garble programs are **compiled to boolean circuits** and always terminate. Garble is **statically typed, low-level, purely functional** and uses a **Rust-like syntax**. Garble is much simpler than Rust though, making it easy to learn and simple to [integrate](./integration.md) into MPC engines.
```
Comment (kisakishy) [inline garble_docs/src/landing.md:3]: See comment on the same paragraph in the ReadMe.

File: README.md [hunk 5]
```diff
@@ -24,36 +22,4 @@ pub fn main(a: u64, b: u64) -> Richest {
...
-## Architecture of this Repository
-
-The Garble compiler is relatively straightforward and turns a program `&str` into a `circuit::Circuit` (or aborts with a scan/parse/type error). The different steps and their modules are as follows (with steps 1-4 happening during compile time, step 5 during run time):
-
-1. [`scan.rs`](src/scan.rs) splits a program `&str` into a `token::Token` sequence.
-2. [`parse.rs`](src/parse.rs) parses a `token::Token` sequence into an untyped `ast::Program`.
-3. [`check.rs`](src/check.rs) type-checks an untyped `ast::Program`, returning a typed `ast::Program`.
-4. [`compile.rs`](src/compile.rs) converts a well-typed `ast::Program` into a `circuit::Circuit`.
-5. [`eval.rs`](src/eval.rs) executes a `circuit::Circuit` with locally supplied inputs.
+To learn more about Garble, check out the [website](https://sine-fdn.github.io/garble-lang)
```
Comment (kisakishy) [inline README.md:25]: Closing dot missing from the end of the sentence.

**Review by kisakishy — COMMENTED** [review]
In general I really like it and the most generic comment I have is about consistency between compiler and language. If we call both simply "Garble", then we could also make that clear in the intro, that Garble is both the name of the tool/compiler and the name of the language as well.

File: garble_docs/src/integration.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
+# MPC Engine Integration
+
+Integrating Garble into your own MPC engine is pretty straightforward. You have to know how to compile Garble programs, how to encode inputs as Garble literals, and how to understand the circuit format that the Garble compiler produces. Let's look at all of these in more detail:
```
[the remark this answers is not in the document]
Comment (fkettelhoit) [reply garble_docs/src/integration.md:3]: I think that's fine though, many other languages with just a single implementation use the name interchangeably for both, the context is normally enough to disambiguate it, I would say.

File: garble_docs/src/guide/installation.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+Commands:
+  run    Run the Garble program with the specified inputs
+  check  Check the Garble program for any type errors
+  help   Print this message or the help of the given subcommand(s)
+
+Options:
+  -h, --help     Print help
+  -V, --version  Print version
+```
```
[the remark this answers is not in the document]
Comment (fkettelhoit) [reply garble_docs/src/guide/installation.md:20]: I think we can assume that if people want to build Garble to integrate it into their own programs they will know enough Rust to be able to use `cargo build`, so I would keep it as it is for now.

**Review by kisakishy — APPROVED** [review]

**Pull request merged**